exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Constraints.kt

```diff
@@ -118,42 +121,36 @@ data class ForeignKeyConstraint(
-                                "Please check your $fromTableName table."
-                        )
-                        is OracleDialect -> exposedLogger.warn(
-                            "Oracle doesn't support FOREIGN KEY with SET DEFAULT reference option with ON DELETE clause. " +
-                                "Please check your $fromTableName table."
-                        )
-                        else -> append(" ON DELETE $deleteRule")
-                    }
+                if (deleteRule == ReferenceOption.RESTRICT && !currentDialect.supportsRestrictReferenceOption) {
+                    exposedLogger.warn(
+                        "${currentDialect.name} doesn't support FOREIGN KEY with RESTRICT reference option with ON DELETE clause. " +
+                            "Please check your $fromTableName table."
+                    )
+                } else if (deleteRule == ReferenceOption.SET_DEFAULT && !currentDialect.supportsSetDefaultReferenceOption) {
+                    exposedLogger.warn(
+                        "${currentDialect.name} doesn't support FOREIGN KEY with SET DEFAULT reference option with ON DELETE clause. " +
+                            "Please check your $fromTableName table."
+                    )
                 } else {
                     append(" ON DELETE $deleteRule")
                 }
             }
+
             if (updateRule != ReferenceOption.NO_ACTION) {
-                if (currentDialect is OracleDialect || currentDialect.h2Mode == H2Dialect.H2CompatibilityMode.Oracle) {
-                    exposedLogger.warn("Oracle doesn't support FOREIGN KEY with ON UPDATE clause. Please check your $fromTableName table.")
-                } else if (updateRule == ReferenceOption.SET_DEFAULT) {
-                    when (currentDialect) {
-                        is MariaDBDialect -> exposedLogger.warn(
-                            "MariaDB doesn't support FOREIGN KEY with SET DEFAULT reference option with ON UPDATE clause. " +
-                                "Please check your $fromTableName table."
-                        )
-                        is MysqlDialect -> exposedLogger.warn(
-                            "MySQL doesn't support FOREIGN KEY with SET DEFAULT reference option with ON UPDATE clause. " +
-                                "Please check your $fromTableName table."
-                        )
-                        else -> append(" ON UPDATE $updateRule")
-                    }
+                if (!currentDialect.supportsOnUpdate) {
+                    exposedLogger.warn("${currentDialect.name} doesn't support FOREIGN KEY with ON UPDATE clause. Please check your $fromTableName table.")
+                } else if (updateRule == ReferenceOption.RESTRICT && !currentDialect.supportsRestrictReferenceOption) {
+                    exposedLogger.warn(
+                        "SQLServer doesn't support FOREIGN KEY with RESTRICT reference option with ON UPDATE clause. " +
```

Comment:
Should this maybe still use `"${currentDialect.name}"`? SQL Server may be the only one that fits the condition but that may change and there isn't a direct dialect check.

Reply:
Yes, this was left by mistake. Thank you!